tests: drivers: dma: dma_m2m_loop build failure fir NXP S32 platforms

### Describe the bug

- s32z2xxdc2@D/s32z270/rtu1
- s32z2xxdc2@D/s32z270/rtu0 
- s32z2xxdc2@B/s32z270/rtu0 
- s32z2xxdc2@B/s32z270/rtu1

8e8056324d355cb70afd35006e27031718fae2cf is the first bad commit
commit 8e8056324d355cb70afd35006e27031718fae2cf
Author: Emilio Benavente <emilio.benavente@nxp.com>
Date:   Wed Jun 18 13:59:42 2025 -0500

    soc: nxp: mcxw: Enable EDMA

    Add DMA nodes for MCXW7X SOC DTS.
    This SOC used TRIGMUX instead of DMAMUX.
    Enable EDMAv3 for the frdm_mcxw71 and frdm_mcxw72
    platforms.

    Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>
    Co-authored-by: Declan Snyder <declan.snyder@nxp.com>

 boards/nxp/frdm_mcxw71/frdm_mcxw71.dts                |  4 ++++
 boards/nxp/frdm_mcxw71/frdm_mcxw71.yaml               |  1 +
 boards/nxp/frdm_mcxw72/frdm_mcxw72_mcxw727c_cpu0.dts  |  4 ++++
 boards/nxp/frdm_mcxw72/frdm_mcxw72_mcxw727c_cpu0.yaml |  1 +
 dts/arm/nxp/nxp_mcxw7x_common.dtsi                    | 19 +++++++++++++++++++
 include/zephyr/dt-bindings/clock/scg_k4.h             |  1 +
 .../hal_nxp/mcux/mcux-sdk-ng/drivers/drivers.cmake    | 16 +++++++++++++++-
 soc/nxp/mcx/mcxw/mcxw7xx/soc.c                        |  4 ++++
 .../dma/loop_transfer/boards/frdm_mcxw71.overlay      |  7 +++++++
 .../boards/frdm_mcxw72_mcxw727c_cpu0.overlay          |  7 +++++++
 tests/drivers/uart/uart_async_api/testcase.yaml       |  2 ++
 11 files changed, 65 insertions(+), 1 deletion(-)
 create mode 100644 tests/drivers/dma/loop_transfer/boards/frdm_mcxw71.overlay
 create mode 100644 tests/drivers/dma/loop_transfer/boards/frdm_mcxw72_mcxw727c_cpu0.overlay


### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b  s32z2xxdc2@D/s32z270/rtu1 tests/drivers/dma/chan_blen_transfer
```

### Relevant log output

```shell
[91/182] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest.c.obj[92/182]
Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_error_hook.c.obj[93/182]
Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_rules.c.obj[94/182]
Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_defaults.c.obj[95/182]
Linking C static library zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a[96/182]
Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_gicv3.c.obj[97/182]
Linking C static library zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a[98/182]
Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_nxp_s32.c.obj[99/182]
Linking C static library zephyr/drivers/clock_control/libdrivers__clock_control.a[100/182]
Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/uart_console.c.obj[101/182]
Linking C static library zephyr/drivers/console/libdrivers__console.a[102/182]
Building C object zephyr/drivers/dma/CMakeFiles/drivers__dma.dir/dma_mcux_edma.c.obj
FAILED: zephyr/drivers/dma/CMakeFiles/drivers__dma.dir/dma_mcux_edma.c.obj zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/drivers/dma/chan_blen_transfer=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/dma/CMakeFiles/drivers__dma.dir/dma_mcux_edma.c.obj -MF zephyr/drivers/dma/CMakeFiles/drivers__dma.dir/dma_mcux_edma.c.obj.d -o zephyr/drivers/dma/CMakeFiles/drivers__dma.dir/dma_mcux_edma.c.obj -c /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/dma/dma_mcux_edma.c/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/dma/dma_mcux_edma.c:28:10: 
fatal error: fsl_dmamux.h: No such file or directory 28 | #include <fsl_dmamux.h> | ^~~~~~~~~~~~~~compilation terminated.ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- os: ubuntu
- toolchain: Zephyr
- version: v4.2.0-6152-gfd51dde8f5ca

### Additional Context

_No response_

--- a/modules/hal_nxp/mcux/mcux-sdk-ng/drivers/drivers.cmake
+++ b/modules/hal_nxp/mcux/mcux-sdk-ng/drivers/drivers.cmake
@@ -304,7 +304,7 @@ if(CONFIG_SOC_SERIES_IMXRT118X)
   set_variable_ifdef(CONFIG_WDT_MCUX_RTWDOG	CONFIG_MCUX_COMPONENT_driver.src_3)
 endif()
 
-if(${MCUX_DEVICE} MATCHES "S32K3")
+if(CONFIG_SOC_SERIES_S32K3 OR CONFIG_SOC_SERIES_S32ZE)
   if(CONFIG_DMA)
     zephyr_include_directories(${MCUX_SDK_NG_DIR}/drivers/dmamux)
     set(CONFIG_MCUX_COMPONENT_driver.dmamux ON)

